File: Mandelbrot_Python_Wrapper_Lab/mandel.c
#include <complex.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>
#include <math.h>


uint8_t* mandelSet(uint8_t* array, int width, int height){
    int xmin = -2;
    int xmax = 1;
    int ymin = -1;
    int ymax = 1;
    double delta_x= (double)(xmax - xmin)/width;
    double delta_y= (double)(ymax-ymin)/height;
    
    for(int i = 0; i < height; i++){
        for(int j = 0; j < width; j++){
            int value = 0;
            double c_real = (xmin + j*delta_x);
            double c_imaginary = (ymax - i*delta_y);

            double c_real_new = c_real;
            double c_imaginary_new = c_imaginary;

            int flag = 0;
            int counter = 0;
            
            double complex mag = c_real_new + c_imaginary_new*I;
            double cmp = cabs(mag);
           
            printf("%f \n", cmp);
            while (cmp < 2.0){
                
                double c_real_new_1 = c_real_new * c_real_new + c_real -c_imaginary_new * c_imaginary_new;
                c_imaginary_new =  2*c_imaginary_new * c_real_new + c_imaginary;
                c_real_new = c_real_new_1;

                mag = c_real_new + c_imaginary_new*I;
                
                cmp = cabs(mag);
                
                if(counter == 100){
                    flag = 1;
                    break;
                }
                counter++;
            }
            array[i*width + j] = flag;
        }
        
    }
    return array;
}

File: Mandelbrot_Python_Wrapper_Lab/mandel.py
import matplotlib.pyplot as plot
import numpy as np

def mandelSet(width, height):
    xmin = -2
    xmax = 1
    ymin = -1
    ymax = 1
    delta_x= (xmax - xmin)/width
    delta_y= (ymax-ymin)/height
    
    array = []
    for i in range(0,height,1):
        for j in range(0,width,1):
            value = 0
            c_real = (xmin + j*delta_x)
            c_imaginary = (ymax - i*delta_y)

            c_real_new = c_real
            c_imaginary_new = c_imaginary

            flag = 0
            counter = 0
            
            mag = np.abs(c_real + c_imaginary*1j)
            print(mag)
            while mag < 2:
                
                c_real_new_1 = c_real_new * c_real_new + c_real -c_imaginary_new * c_imaginary_new
                c_imaginary_new =  2*c_imaginary_new * c_real_new + c_imaginary
                c_real_new = c_real_new_1

                mag = np.abs(c_real_new + c_imaginary_new*1j)

                if counter == 100:
                    flag = 1
                    break
                counter = counter + 1

            array.append(flag)
    return array


if __name__ == "__main__":

    w = 800
    h = 700
    
    array = mandelSet(w, h)

    plotar = np.reshape(array, (h, w))
    plot.imshow(plotar)
    plot.show()